app/models/manageiq/providers/autosde/inventory/collector/storage_manager.rb

```diff
@@ -26,6 +25,44 @@ def storage_resources
     end
   end
 
+  def san_addresses
+    @san_addresses ||= @manager.autosde_client.StorageHostApi.storage_hosts_get.map do |addresses|
+      addresses_array = []
+
+      addresses.addresses.each do |address|
```

Comment:
Instead of a local array, appending to the array on each loop iteration, and returning the array...this can be simplified by using `#map` 

```
addresses.addresses.map do |address|
  {
    :ems_ref => address.uuid,
    ...
  }
end
```

Map will automatically build an array based on the return from the block and return it

Reply:
changed like you suggested, only with `flat_map`